lib/Ptero/Builder/Workflow.pm

```diff
@@ -43,33 +43,29 @@ override 'BUILDARGS' => sub {
     return $params;
 };
 
-sub submit_url {
-    if (exists $ENV{PTERO_WORKFLOW_SUBMIT_URL}) {
-        return $ENV{PTERO_WORKFLOW_SUBMIT_URL};
-    } else {
-        die "environment variable PTERO_WORKFLOW_SUBMIT_URL must be set";
-    }
-}
-
 sub submit {
     my $self = shift;
     my %p = Params::Validate::validate(@_, {
         inputs => {type => HASHREF, optional => 1},
         name => {type => SCALAR, optional => 1},
+        submit_url => {type => SCALAR, optional => 1},
     });
 
+    my $submit_url = $p{submit_url} || $ENV{PTERO_WORKFLOW_SUBMIT_URL} || die
+        "Must specify 'submit_url' or set PTERO_WORKFLOW_EXECUTION_URL env variable.";
```

Comment:
The env variable named in this error message doesn't match the variable that is checked.